build_tools/docker/context/install_ccache.sh

```diff
@@ -7,16 +7,25 @@
 # SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 
 CCACHE_VERSION=4.7.4
+machine=$(uname -m)
+if [[ $machine == x86_64 ]]; then
+  curl --silent --show-error --fail --location \
+      "https://github.com/ccache/ccache/releases/download/v${CCACHE_VERSION}/ccache-${CCACHE_VERSION}-linux-x86_64.tar.xz" \
+      --output ccache.tar.xz \
+      "https://github.com/ccache/ccache/releases/download/v${CCACHE_VERSION}/ccache-${CCACHE_VERSION}-linux-x86_64.tar.xz.asc" \
+      --output ccache.tar.xz.asc \
+      "https://keyserver.ubuntu.com/pks/lookup?op=get&search=0x5a939a71a46792cf57866a51996dda075594adb8" \
+      --output JOEL_PUBKEY.asc
+  gpg --import JOEL_PUBKEY.asc
+  gpg --verify ccache.tar.xz.asc
+  tar -xvf ccache.tar.xz --strip-components=1
+  cp ccache /usr/bin/
+elif  [[ $machine == aarch64 ]]; then
+  # Apt install for Ubuntu 20.04 gives ccache version 3.7.7, I assume you want
+  # something newer. Two alternatives from what I can find
+  #   1) archlinuxarm.org has a 4.8.2 version
+  #   2) build from source
+  # but I may be missing something...
+  apt-get install -y ccache
```

Comment:
Question as stated in the comments. LMKWYT

Reply:
I have version 4.7.5 on one of my machines. I don't think we're too sensitive to the ccache version... we could look through https://ccache.dev/releasenotes.html to see what we'd be missing out on by using a version as old as 3.7.7 (2020-01-05: https://ccache.dev/releasenotes.html#_ccache_3_7_7).

* 4.0
  * changed the compression algorithm
  * added the ability to recompress at a higher level (we've tried this as a way to build at a low compression level, recompress at a higher level, then persist the cache in a GitHub Actions cache)
* 4.1
  * follow symlinks when guessing the compiler (could be useful if we aren't explicit)
* 4.2
  * better CUDA/clang support (we might notice some cache misses without that?)
* 4.5
  * `CCACHE_NAMESPACE` -- this we probably want, as we use namespaces quite a bit
* 4.6
  * Support for MSVC (we want this for Windows builds... not sure if that matters in this context though)

Reply:
Huh https://ccache.dev/platform-compiler-language-support.html doesn't say anything about architecture support, but then all the releases are only for X86 :-(

Reply:
Yeah, 3.7.7 to 4.7.5 is a big leap. How about we use 3.7.7 via `apt-get install -y ccache` or the 4.8.2 via archlinuxarm.org for now? Do you have a preference?

I can look into arm64 support on ccache.dev, but that might take a while...